runtime/core.go

```diff
@@ -493,13 +493,14 @@ func IndexInt(f *Frame, o *Object) (int, *BaseException) {
 	if raised != nil {
 		return 0, raised
 	}
-	if i != nil {
-		if i.isInstance(IntType) {
-			return toIntUnsafe(i).Value(), nil
-		}
-		if l := toLongUnsafe(i).Value(); numInIntRange(l) {
-			return int(l.Int64()), nil
-		}
+	if i == nil || (!i.isInstance(IntType) && !i.isInstance(LongType)) {
```

Comment:
I think things can be simplified somewhat because int and long both have the `__index__` slot. I believe Index() checks whether o is int or long as an optimization, but perhaps it's better if we just remove the check on 469.

That will guarantee that either we already raised or `i == nil || i.isInstance(IntType) || i.isInstance(LongType))` so we can remove the second part of the condition here.